tests/unit/amo/pages/TestAddon.js

```diff
@@ -1632,4 +1633,18 @@ describe(__filename, () => {
       expect(root.find(InstallWarning)).toHaveProp('addon', internalAddon);
     });
   });
+
+  it('displays a link for a QR code with an add-on', () => {
+    const addon = createInternalAddonWithLang(fakeAddon);
+    const root = shallowRender({ addon });
+
+    expect(root.find(AddonQRCodeLink)).toHaveLength(1);
+    expect(root.find(AddonQRCodeLink)).toHaveProp('addon', addon);
+  });
+
+  it('does not display a link for a QR code without an add-on', () => {
+    const root = shallowRender({ addon: null });
```

Comment:
We can omit this test case if `AddonQRCodeLink` accepts an optional `addon` prop.